ipa-deploy/orb.yml

```diff
@@ -79,16 +91,217 @@ commands:
-
       - run:
           name: Export template to environment variable
           command: |
             echo $(cat /tmp/ipa-deploy/failed_deploy_template.json)
             echo 'export SLACK_DEPLOY_FAILED_TEMPLATE=$(cat /tmp/ipa-deploy/failed_deploy_template.json)' >> $BASH_ENV
+  
+  npm_ci:
+    description: Install node dependencies
+    parameters:
+      executor:
+        type: string
+        description: The executor used. Needed to avoid conflicting caches
+    steps:
+      - restore_cache:
+          keys:
+            - v1-dependencies-"<<parameters.executor>>"-{{ "package.json" }}
+            - v1-dependencies-
+      - run:
+          name: "Set NPM token"
+          command: echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > ~/.npmrc
+      - run:
+          name: "Install NPM dependencies."
+          command: |
+            npm ci
+      - save_cache:
+          paths:
+            - node_modules
+          key: v1-dependencies-"<<parameters.executor>>"-{{ checksum "package.json" }}
+  
+  notify-slack-on-failure:
+    description: Sends a slack notification if the job fails
+    parameters:
+      channel:
+        type: string
+        description: The channel to send the notification to
+      mentions:
+        type: string
+        description: The Slack users/groups to mention in the message
+        default: ""
+
+    steps:
+      - attach_workspace:
+          at: .
+      - load_templates
+      - slack/notify:
+          event: fail
+          channel: <<parameters.channel>>
+          mentions: <<parameters.mentions>>
+          template: SLACK_DEPLOY_FAILED_TEMPLATE
+  
+  deploy-run-integration-tests:
+    description: Run integration tests
+    parameters:
+      executor:
+        type: string
+        description: The executor used. Needed to avoid conflicting caches
+      stage:
+        type: string
+        default: integration
+      region:
+        default: ""
+        type: string
+    steps:
+      - attach_workspace:
+          at: .
+      - restore_cache:
+          keys:
+            - v1-dependencies-"<<parameters.executor>>"-{{ "package.json" }}
+            - v1-dependencies-
+      - run:
+          name: "Set NPM token"
+          command: echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > ~/.npmrc
+      - run:
+          name: "Install NPM dependencies."
+          command: |
+            npm ci
+      - run:
+          name: "Install serverless CLI"
+          command: sudo npm install -g serverless@3.11.0
+      - run:
+          command: serverless plugin install -n serverless-plugin-aws-alerts
+          name: Install serverless-aws-alerts
+      - run:
+          command: |
+            export DEPLOYMENT_TYPE=integrationTest
+            serverless deploy --stage ${STAGE} --region ${AWS_REGION}
+          name: deploy
+      - run:
+          name: Checking env variable
+          command: echo ${KTR_REQUEST_INPUT_QUEUE}
```

Comment:
specific only to KTR?

Reply:
oops was just using this for debugging will remove